### ISSUE TITLE
Wrong error when trying to create group with existing name

### ISSUE THREAD

**Issue by Dimon4eg** [heroiclabs/nakama#357]
## Description
Wrong error code 13 is received from server when trying to create group with existing name.

## Steps to Reproduce
1. Create group with name A
2. Try to create group with name A again
3. Check error

## Expected Result
Should be error code 6 - Already exists

## Actual Result
{"error":"Error while trying to create group.","message":"Error while trying to create group.","code":13}

## Context
C++ client 2.1.0

## Your Environment
- Nakama: 2.4.0
- Cockroach: 2.1.4

**Comment by mofirouz** [comment]
@Dimon4eg Thanks for this - can you do me a favour:

1. Repeat this experiment with Nakama 2.5.1 on Cockroach v2.1.7
2. Post the server logs as soon as you get error 13.

Thanks.

**Comment by Dimon4eg** [comment]
The same error with Nakama server 2.5.1:
```
{"level":"debug","ts":"2019-06-30T15:54:02.043+0300","msg":"Could not create group.","error":"pq: duplicate key value (name)=(e'We\\'re-Nakama-Lovers') violates unique constraint \"groups_name_key\""}
{"level":"error","ts":"2019-06-30T15:54:02.045+0300","msg":"Error creating group.","error":"pq: duplicate key value (name)=(e'We\\'re-Nakama-Lovers') violates unique constraint \"groups_name_key\"","stacktrace":"github.com/heroiclabs/nakama/server.CreateGroup\n\t/go/src/github.com/heroiclabs/nakama/server/core_group.go:132\ngithub.com/heroiclabs/nakama/server.(*ApiServer).CreateGroup\n\t/go/src/github.com/heroiclabs/nakama/server/api_group.go:57\ngithub.com/heroiclabs/nakama/apigrpc._Nakama_CreateGroup_Handler.func1\n\t/go/src/github.com/heroiclabs/nakama/apigrpc/apigrpc.pb.go:1150\ngithub.com/heroiclabs/nakama/server.StartApiServer.func1\n\t/go/src/github.com/heroiclabs/nakama/server/api.go:101\ngithub.com/heroiclabs/nakama/apigrpc._Nakama_CreateGroup_Handler\n\t/go/src/github.com/heroiclabs/nakama/apigrpc/apigrpc.pb.go:1152\ngithub.com/heroiclabs/nakama/vendor/google.golang.org/grpc.(*Server).processUnaryRPC\n\t/go/src/github.com/heroiclabs/nakama/vendor/google.golang.org/grpc/server.go:971\ngithub.com/heroiclabs/nakama/vendor/google.golang.org/grpc.(*Server).handleStream\n\t/go/src/github.com/heroiclabs/nakama/vendor/google.golang.org/grpc/server.go:1250\ngithub.com/heroiclabs/nakama/vendor/google.golang.org/grpc.(*Server).serveStreams.func1.1\n\t/go/src/github.com/heroiclabs/nakama/vendor/google.golang.org/grpc/server.go:690"}
```

**Comment by zyro** [comment]
Fixed in 4ce767230523e84f6a181c6b0ce943c1e59d626f and will be part of the next release.
